Modules/_threadmodule.c

```diff
@@ -2403,7 +2403,10 @@ of the main interpreter.");
 #   include <pthread_np.h>
 #endif
 
-#if defined(HAVE_PTHREAD_GETNAME_NP) || defined(HAVE_PTHREAD_GET_NAME_NP) || defined(MS_WINDOWS)
+
+#if (((defined(HAVE_PTHREAD_GETNAME_NP) || defined(HAVE_PTHREAD_GET_NAME_NP)) \
+      && defined(_PYTHREAD_NAME_MAXLEN)) \
```

Comment:
Isn't this a regression? If `_PYTHREAD_NAME_MAXLEN` is not defined, there was still a chance to get a name if it was shorter than 100 bytes.